Components/StorageHeart.cs

```diff
@@ -23,17 +23,17 @@ public override TEStorageHeart GetHeart(int i, int j)
 		public override bool RightClick(int i, int j)
 		{
 			Player player = Main.LocalPlayer;
-			Item item = player.HeldItem;
-			if (item.type == ModContent.ItemType<Locator>() || item.type == ModContent.ItemType<LocatorDisk>() || item.type == ModContent.ItemType<PortableAccess>())
+			ModItem item = player.HeldItem.ModItem;
+
+			if (item != null && item is Locator locator)
```

Comment:
Null check isn't needed, but a bit of extra safety can't hurt.
Change this to if (`!item.IsAir && item.ModItem is Locator locator)`